### DISCUSSION TITLE
How can I use onedrive on OpenWRT?

### DISCUSSION THREAD

**Discussion by sgpublic** [abraunegg/onedrive#1923] [Q&A] [answered]
I found that the executable is not given here, but requires users to compile and install it on their own systems. But this seems difficult to implement on OpenWRT, and I'm not very clear on exactly which dependencies are required.

What do I need to do if I want to use it on OpenWRT?

**Comment by abraunegg** [comment, marked as answer]
@sgpublic 
> What do I need to do if I want to use it on OpenWRT?

Your options are going to be very limited

**Option 1:**
1. Review https://openwrt.org/docs/guide-developer/toolchain/single.package and step through the client dependencies that are required to compile - for instance libcurl, libsqlite3, git, curl for starters, but there could be a few more based on your choice of platform. I do not even know if they are available on your platform, or what even any of the compiler dependencies may be
2. Review if there is an LDC compiler available for your CPU architecture - https://github.com/ldc-developers/ldc/releases and install that and confirm that you are able to run the actual compiler
3. Ensure you meet the compiler requirements of 1GB memory + 1GB swap space otherwise building the client will fail

**Option 2:**
1. Review https://openwrt.org/docs/guide-user/virtualization/docker_host and install Docker 
2. Compile the required Docker container to run - do not pull & run from Docker as the images are out-of-date due to Docker closing down GitHub integration

**Option 3:**
1. Submit a feature request to the OpenWRT team to include the client in their distribution

**Option 4:**
1. Dont run / use OpenWRT, get a better solution that supports more options
